metadata-models/src/main/pegasus/com/linkedin/join/FieldMap.pdl

```diff
@@ -0,0 +1,19 @@
+namespace com.linkedin.join
+
+import com.linkedin.dataset.SchemaFieldPath
+
+/**
+ * Field Mapping of 1:1 field
+ */
+record FieldMap {
+  /**
+   * All fields from dataset A that are required for the join, maps to bFields 1:1
+   */
+  afield: SchemaFieldPath
```

Comment:
Could we call this `leftDatasetFields` and `rightDatasetFields`, or something else that is more descriptive? I think these names are somewhat confusing